substreams/ethereum-maverick/src/modules.rs

```diff
@@ -0,0 +1,456 @@
+                                from: tx.from.clone(),
+                                to: tx.to.clone(),
+                                index: Into::<u64>::into(tx.index),
+                            }),
+                            tokens: vec![pool_added.token_a, pool_added.token_b],
+                            contracts: vec![FACTORY.into(), pool_added.pool_address],
+                            static_att: vec![
+                                Attribute {
+                                    name: "activeTick".into(),
+                                    value: pool_added
+                                        .active_tick
+                                        .to_signed_bytes_be(),
+                                    change: ChangeType::Creation.into(),
+                                },
+                                Attribute {
+                                    name: "lookback".into(),
+                                    value: pool_added.lookback.to_signed_bytes_be(),
+                                    change: ChangeType::Creation.into(),
+                                },
+                            ],
+                            change: ChangeType::Creation.into(),
+                            protocol_type: Some(ProtocolType {
+                                name: "maverick_pool".into(),
+                                financial_type: FinancialType::Swap.into(),
+                                attribute_schema: Vec::new(),
+                                implementation_type: ImplementationType::Vm.into(),
+                            }),
+                        })
+                    })
+                    .collect::<Vec<_>>();
+
+                if !components.is_empty() {
+                    Some(TransactionProtocolComponents {
+                        tx: Some(Transaction {
+                            hash: tx.hash.clone(),
+                            from: tx.from.clone(),
+                            to: tx.to.clone(),
+                            index: Into::<u64>::into(tx.index),
+                        }),
+                        components,
+                    })
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>(),
+    })
+}
+
+/// Simply stores the `ProtocolComponent`s with the pool id as the key
+#[substreams::handlers::store]
+pub fn store_components(map: BlockTransactionProtocolComponents, store: StoreAddInt64) {
+    store.add_many(
+        0,
+        &map.tx_components
+            .iter()
+            .flat_map(|tx_components| &tx_components.components)
+            .map(|component| format!("pool:{0}", component.id))
+            .collect::<Vec<_>>(),
+        1,
+    );
+}
+
+/// Simply stores the `ProtocolComponent`s with the pool id as the key
+#[substreams::handlers::store]
+pub fn store_component_tokens(map: BlockTransactionProtocolComponents, store: StoreSetString) {
```

Comment:
I think it'd be better to use a `StoreSetRaw` like: https://github.com/propeller-heads/tycho-protocol-sdk/pull/142/files#diff-08d942a7252e19f62048ce685378d53cd696e37570289ca900ae956d24e0619dR91